vars/Debian-10.yml

```diff
@@ -1,12 +1,11 @@
 ---
-__postgresql_version: "11"
+__postgresql_version: "{{ postgresql_version | default('11') }}"
```

Comment:
I don't believe you need to change all these particular defaults (in this file and the others), since setting `postgresql_version` itself will override the default via the vars setup task I have in this role.

Reply:
If I don't change it, there's an error of `postgresql_version` not being defined ( sort of chicken-egg problem), that's why I had to resort to change the way the variable works.

Reply:
FYI I am quite certain that you cannot refer to lower-precedence variables (such as those defined in `defaults/...`  inside high-raking precedence defined in `vars/...`. Effectively, `__postgresql_version` will always be the default '11', and `postgresql_version` will be considered non-existent here because not defined at this level of precedence (see ansible variable precedence).

Reply:
I might be wrong, but if you pass the var to the role, it's already defined at this point in time, so the `|` will take `postgresql_version`, since it's define way earlier than the role vars are. so, the end result is that you get the default if `postgresql_version` is undefined, and later on `postgresql_version` is set to the default if it was undefined before that.

Reply:
Hmm you might be right. It was one of the first thing I tried, but got an error without the `default` filter. Though I didn't look too much into it, I thought it was because of [variable precedence rules](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable). Role vars are n°15 (high precedence), where role defaults are only n°2 (low precedence). But role (and include_role) params are n°20 so I guess if you feed them upon role inclusion that should over-ride everything?
I ended up leveraging the way `tasks/variables.yml` works here and fed all the OS-specific variables to the role directly.
https://github.com/geerlingguy/ansible-role-postgresql/blob/1f873d304b5b91b9eb81e24b6d7fce4c584a61e4/tasks/variables.yml#L17-L21
I also noticed, for example, that changing the data_dir variable does not change it in PostgreSQL's `.conf` (**not sure this is intended behavior?**), so I had to also pass that 'manually' via the role's `postgresql_global_config_options` var.
In any case, wouldn't just modding the Ubuntu/Debian vars file make the api of the role inconsistent across platforms?

Reply:
>if you feed them upon role inclusion that should over-ride everything?
To the extent that I have observed this, yes. 

> I also noticed, for example, that changing the data_dir variable does not change it in PostgreSQL's .conf (not sure this is intended behavior?),
My guess would be that's a bug.